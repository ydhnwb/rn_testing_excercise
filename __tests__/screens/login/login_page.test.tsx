import React from 'react';
import renderer from 'react-test-renderer'
import LoginPage from '../../../src/presentation/screen/login/login_page';
import { render, fireEvent } from '@testing-library/react-native';


describe("test the login page", () => {
    it("should render correctly", () => {
        const widget = renderer.create(<LoginPage/>)
        expect(widget).toMatchSnapshot()
    })

    //gonna use render, not renderer
    it("validation should run correctly", () => {
        const { getByTestId, getByPlaceholderText, getByText, queryByText } = render(<LoginPage/>)

        //it is actually like doing web crawling.
        //also you can use by testId so it get the component directly, BUT it you need to pass ...props to every component
        const loginButton = getByTestId("buttonLogin")

        //should show email error
        fireEvent.press(loginButton)
        getByText("Email is not valid")

        //should show password error AND error email SHOULD NOT BE SHOWN
        fireEvent.changeText(getByPlaceholderText("Email"), "yudhanewbie@gmail.com")
        fireEvent.press(loginButton)
        getByText("Password is not valid")
        expect(queryByText("Email is not valid")).toBe(null)
    })
})