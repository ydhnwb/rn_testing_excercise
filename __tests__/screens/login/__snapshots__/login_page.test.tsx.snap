// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the login page should render correctly 1`] = `
<View
  style={
    Object {
      "padding": 16,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "600",
      }
    }
  >
    Hello login
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    underlineColorAndroid="blue"
  />
  <Text
    style={
      Array [
        Object {
          "color": "red",
        },
        Object {
          "marginBottom": 4,
          "marginTop": 2,
        },
      ]
    }
    testID="textEmailError"
  >
    
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    underlineColorAndroid="blue"
  />
  <Text
    style={
      Array [
        Object {
          "color": "red",
        },
        Object {
          "marginBottom": 4,
          "marginTop": 2,
        },
      ]
    }
    testID="textPasswordError"
  >
    
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="buttonLogin"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Sign in
      </Text>
    </View>
  </View>
</View>
`;
